_posts/2018-12-14-Bazel-0.20.0.md

```diff
@@ -0,0 +1,92 @@
+ 
+Breaking changes in the next release (0.21 )[are marked with GitHub label 'breaking-change-0.21'](https://github.com/bazelbuild/bazel/issues?q=is%3Aissue+label%3Abreaking-change-0.21)
+
+## General Changes
+
+  - (Linux only) Bazel will now shut down when idle for 5 minutes and the system
+    is low on RAM.
+  - We're deprecating --genrule_strategy=<value> with the intent of removing it in early 2019; please use `--strategy=Genrule=<value>` instead
+    (see [#6760](https://github.com/bazelbuild/bazel/issues/6760)).
+  - Support for querying remote server Capabilities (per REAPI v2)
+  - Use different server log files per Bazel server process; java.log
+    is now a symlink to the latest log.
+  - The code coverage report now includes the actual paths to header
+    files instead of the ugly, Bazel generated, virtual includes path.
+  - Sorting remote Platform properties for remote execution. 
+
+## C++
+
+  - The --experimental_no_dotd_scanning_with_modules command line argument is not supported anymore.
+  - The --prune_cpp_modules command line option is not supported anymore.
+  - The --experimental_prune_cpp_input_discovery command line option is not supported anymore.
+  - CppRules: All cc_toolchains depended on from cc_toolchain_suite.toolchains are now analyzed when not using
+    platforms in order to select the right cc_toolchain.
+  - CROSSTOOL file is now read from the package of cc_toolchain, not from
+    the package of cc_toolchain_suite. This is not expected to break anybody since
+    cc_toolchain_suite and cc_toolchain are commonly in the same package.
+  - CppRules: All cc_toolchains depended on from
+    cc_toolchain_suite.toolchains are now analyzed when not using
+    platforms in order to select the right cc_toolchain.
+    
+## Java
+
+  - removed obsolete `--explicit_jre_deps` flag.
+  - If the `--javabase` flag is unset, Bazel locates a JDK using the `JAVA_HOME` environment variable 
+    and searching the PATH. If no JDK is found, `--javabase` will be empty and builds targeting Java
+    will not be supported. Previously Bazel would fall back to using the embedded  JDK as a --javabase,
+    but this is no longer default behaviour. A JDK should be explicitly installed instead to enable Java development.
+
+## Android
+
+  - Added support for Android NDK r18.
+  - Make legacy-test-support ("legacy_test-<api-level>") from
+    android_sdk_repository neverlink. The legacy test support
+    libraries shouldn't be built into test binaries. To make them
+    available at runtime, developers should declare them via
+    uses-library:
+    https://developer.android.com/training/testing/set-up-project#android-test-base
+    
+ ## Starlark
+  - All overrides of Starlark's ctx.new_file function are now
+    deprecated. Try the `--incompatible_new_actions_api` flag to ensure your
+    code is forward-compatible.
+  - The function `attr.license` is deprecated and will be removed. It can be disabled now
```

Comment:
Just say "has been deprecated." If you say "will be removed" the first question the user asks is "when?" and we should not provide any promises for future functionality changes.